src/System.Reflection.Metadata/src/System/Reflection/Metadata/Handles.cs

```diff
@@ -7,77 +7,179 @@
 
 namespace System.Reflection.Metadata
 {
+    /// <summary>
+    /// Represents any metadata entity (type reference/definition/specification, method definition, custom attribute, etc.) or value (string, blob, guid, user string).
+    /// </summary>
+    /// <remarks>
+    /// Use <see cref="Handle"/> to store multiple kinds of handles.
+    /// </remarks>
     public struct Handle : IEquatable<Handle>
     {
-        internal readonly uint value;
+        private readonly int _value;
 
-        internal Handle(uint value)
+        // bits:
+        //    7: IsVirtual
+        // 0..6: token type
+        private readonly byte _vType;
```

Comment:
Because there are APIs that operate on any handle and they are actually useful. We'd need to have 4 separate overloads for each such API and the callers would need to switch on the handle kind every time they use these APIs.